### PR TITLE
feat: integrate material theme with themebuilder

### PR DESCRIPTION
**Pull request by joneff**
Initial changes so material theme can be integrated in themebuilder.

Needs #471

### CODE REVIEW

**Comment by joneff** [comment]
Ideally, this is intended to be used by customers, but the first stage is to have the ability to integrate material theme in themebuilder.

This is related to the offline discussion we had about #366, especially the swatches part.

File: packages/material/scss/_swatches.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+$surface-name: light !default; // light or dark
+
+
+$primary-palette: get-palette($primary-name);
+$secondary-palette: get-palette($secondary-name);
+
+
+$primary: map-get($primary-palette, 500);
+$primary-darker: map-get($primary-palette, 700);
+$primary-lighter: map-get($primary-palette, 200);
```
Comment (gyoshev) [inline packages/material/scss/_swatches.scss]: These appear to be the main reason for using the named palettes, but they are not used anywhere in the theme. Perhaps these colors need to be set to other variables?

File: packages/material/scss/_swatches.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+$secondary: map-get($secondary-palette, A200);
+
+
+$surface: if($surface-name == light, #ffffff, #303030);
```
[the remark this answers is not in the document]
Comment (joneff) [reply packages/material/scss/_swatches.scss]: I've switched to `$theme-type: light | dark` which makes more sense

File: packages/material/scss/_swatches.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+$primary-name: indigo !default;
+$secondary-name: pink !default;
+$surface-name: light !default; // light or dark
+
+
+$primary-palette: get-palette($primary-name);
+$secondary-palette: get-palette($secondary-name);
+
+
+$primary: map-get($primary-palette, 500);
```
[the remark this answers is not in the document]
Comment (joneff) [reply packages/material/scss/_swatches.scss]: Done. I've `!default`-ed the variables.

**Comment by joneff** [comment]
@ivanchev, the initial idea was to "just" add material theme in themebuilder, but that didn't work out:

-- there were changes needed to be made in themebuilder, which we already did
-- the actual changes done to material theme are beyond what we initially thought

File: packages/material/scss/_swatches.scss (hunk 1 above)
Comment (joneff) [reply packages/material/scss/_swatches.scss]: `$primary-lighter` is use in progressbar

**Pull request merged**